tests/test_file_operations.py

```diff
@@ -0,0 +1,172 @@
+    assert content == new_content
+
+
+def test_append_to_file(test_nested_file):
+    append_text = "This is appended text.\n"
+    write_to_file(test_nested_file, append_text)
+
+    append_to_file(test_nested_file, append_text)
+
+    with open(test_nested_file, "r") as f:
+        content_after = f.read()
+
+    assert content_after == append_text + append_text
+
+
+def test_delete_file(config, test_file):
+    delete_file(test_file)
+    assert os.path.exists(test_file) is False
+    assert delete_file(test_file) == "Error: File has already been deleted."
+
+
+def test_delete_missing_file(test_file):
+    os.remove(test_file)
+    try:
+        os.remove(test_file)
+    except FileNotFoundError as e:
+        error_string = str(e)
+        assert error_string in delete_file(test_file)
+        return
+    assert True, "Failed to test delete_file"
+
+
+def test_search_files(config, workspace, test_directory):
+    # Case 1: Create files A and B, search for A, and ensure we don't return A and B
+    file_a = workspace.get_path("file_a.txt")
+    file_b = workspace.get_path("file_b.txt")
+
+    with open(file_a, "w") as f:
+        f.write("This is file A.")
+
+    with open(file_b, "w") as f:
+        f.write("This is file B.")
+
+    # Create a subdirectory and place a copy of file_a in it
+    if not os.path.exists(test_directory):
+        os.makedirs(test_directory)
+
+    with open(os.path.join(test_directory, file_a.name), "w") as f:
+        f.write("This is file A in the subdirectory.")
+
+    files = search_files(str(workspace.root))
+    assert file_a.name in files
+    assert file_b.name in files
+    assert os.path.join(Path(test_directory).name, file_a.name) in files
+
+    # Clean up
+    os.remove(file_a)
+    os.remove(file_b)
+    os.remove(os.path.join(test_directory, file_a.name))
+    os.rmdir(test_directory)
+
+    # Case 2: Search for a file that does not exist and make sure we don't throw
+    non_existent_file = "non_existent_file.txt"
+    files = search_files("")
+    assert non_existent_file not in files
+
+
+def test_download_file():
+    url = "https://github.com/Significant-Gravitas/Auto-GPT/archive/refs/tags/v0.2.2.tar.gz"
+    local_name = os.path.join(gettempdir(), "auto-gpt.tar.gz")
+    size = 365023
+    readable_size = readable_file_size(size)
+    assert (
+        download_file(url, local_name)
+        == f'Successfully downloaded and locally stored file: "{local_name}"! (Size: {readable_size})'
+    )
+    assert os.path.isfile(local_name) is True
+    assert os.path.getsize(local_name) == size
+
+    url = "https://github.com/Significant-Gravitas/Auto-GPT/archive/refs/tags/v0.0.0.tar.gz"
+    assert "Got an HTTP Error whilst trying to download file" in download_file(
+        url, local_name
+    )
+
+    url = "https://thiswebsiteiswrong.hmm/v0.0.0.tar.gz"
+    assert "Failed to establish a new connection:" in download_file(url, local_name)
+
+
+if __name__ == "__main__":
+    pytest.main()
```

Comment:
This is not necessary

Reply:
Took if from another test file. :angel: Removed.